assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,304 @@
+  ]);
+
+  useEffect(() => {
+    inputRef.current.focus();
+  });
+
+  useEffect(() => {
+    const listElement = listContainerRef.current;
+
+    const handleScroll = () => {
+      let { scrollTop } = listElement;
+      scrollTop -= 5;
+      let currentVisibleIndex = Math.floor(scrollTop / 34);
+      if (
+        recentUsedFonts.length > 0 &&
+        currentVisibleIndex > recentUsedFonts.length
+      ) {
+        scrollTop -= 10;
+        currentVisibleIndex =
+          Math.floor(scrollTop / 34) - recentUsedFonts.length;
+      }
+      if (normalFonts.length > 0 && currentVisibleIndex > normalFonts.length) {
+        scrollTop -= 10;
+      }
+      currentVisibleIndex = Math.floor(scrollTop / 34);
+      if (currentVisibleIndex < 0) {
+        currentVisibleIndex = 0;
+      }
+      if (currentVisibleIndex !== currentActiveRef.current) {
+        currentActiveRef.current = currentVisibleIndex;
+        setCurrentIndex(currentVisibleIndex);
+      }
+    };
+    listElement.addEventListener('scroll', handleScroll, { passive: false });
+
+    return () => {
+      listElement.removeEventListener('scroll', handleScroll);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useFocusOut(pickerContainerRef, toggleOptions);
+
+  useKeyDownEffect(pickerContainerRef, { key: 'esc' }, toggleOptions, [
+    toggleOptions,
+  ]);
+
+  const handleItemClick = (option, slug) => {
+    handleCurrentValue(option);
+    addUsedFont(slug);
+  };
+
+  const renderListWithOptions = (options, offset = 0) => {
+    if (options?.length > 0) {
+      return (
+        <List aria-multiselectable={false} aria-required={false} ref={listRef}>
+          {options.map(({ name, value, slug }, index) => (
+            <Item
+              key={value}
+              onClick={() => handleItemClick(value, slug)}
+              fontFamily={
+                index < currentIndex + 10 - offset &&
+                index >= currentIndex - offset
+                  ? name
+                  : undefined
+              }
+            >
+              {name}
+            </Item>
+          ))}
+        </List>
+      );
+    }
+    return null;
+  };
+
+  return (
+    <PickerContainer ref={pickerContainerRef}>
+      <ExpandedTextInput
+        forwardedRef={inputRef}
+        value={searchValue}
+        onChange={setSearchValue}
+        color="white"
+        clear
+      />
+      <ListContainer
+        ref={listContainerRef}
+        aria-labelledby={__('FontPicker', 'web-stories')}
+      >
+        <GlobalFontFaces menuFonts={menuFonts.join('')} />
+        {renderListWithOptions(recentUsedFonts)}
+        {renderListWithOptions(normalFonts, recentUsedFonts.length)}
+        {renderListWithOptions(
+          includeSearchFonts,
+          normalFonts.length + recentUsedFonts.length
+        )}
+        {recentUsedFonts.length === 0 &&
+          normalFonts.length === 0 &&
+          includeSearchFonts.length === 0 && (
```

Comment:
You can use `!recentUsedFonts.length && ...` for more readability. WDYT?

Reply:
Agree. Fixed!